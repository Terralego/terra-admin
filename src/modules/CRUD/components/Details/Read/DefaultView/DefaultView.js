import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { Tab, Tabs } from '@blueprintjs/core';
import { NavLink, useParams } from 'react-router-dom';

import { generateURI } from '../../../../config';

import PropertyList from '../../PropertyList';

const DefaultView = ({ displayProperties, properties }) => {
  const tabs = useMemo(() => Object.values(displayProperties)
    .sort((a, b) => a.order - b.order),
  [displayProperties]);

  const {
    category,
    id,
    layer,
    section = 'default',
  } = useParams();

  const editProperties = useMemo(() => (
    Object.values(properties).reduce((list, item) => ({ ...list, ...item }), {})
  ), [properties]);

  const { t } = useTranslation();

  if (!tabs.length) {
    return null;
  }

  return (
    <Tabs
      selectedTabId={category || tabs[0].slug}
    >
      {tabs.map(({ title, slug = 'other', properties: propertiesFromTab }) => (
        <Tab
          key={slug}
          id={slug}
          title={<NavLink to={generateURI('layer', { layer, id, section, category: slug })}>{title || t('CRUD.details.other')}</NavLink>}
          panel={<PropertyList properties={propertiesFromTab} editProperties={editProperties} />}
        />
      ))}
      <Tabs.Expander />
    </Tabs>
  );
};

DefaultView.propTypes = {
  displayProperties: PropTypes.shape({}),
  properties: PropTypes.shape({}),
};

DefaultView.defaultProps = {
  displayProperties: {},
  properties: {},
};

export default DefaultView;
