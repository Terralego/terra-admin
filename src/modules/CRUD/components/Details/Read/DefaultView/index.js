import DefaultView from './DefaultView';

export default DefaultView;
