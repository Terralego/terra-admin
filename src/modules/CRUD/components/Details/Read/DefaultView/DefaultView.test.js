import React from 'react';
import renderer from 'react-test-renderer';

import DefaultView from './DefaultView';

jest.mock('@blueprintjs/core', () => {
  const Tabs = ({ children }) => <ul>{children}</ul>;
  const Tab = ({ title, panel }) => <li>{title}{panel}</li>;
  Tabs.Expander = () => null;

  return {
    Tabs,
    Tab,
  };
});

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: text => text,
  }),
}));

jest.mock('react-router-dom', () => ({
  NavLink: ({ children }) => <span>{children}</span>,
  useParams: () => ({
    category: 'someCategory',
    id: 'layerId',
    layer: 'layerFoo',
    section: 'default',
  }),
}));

jest.mock('../../PropertyList', () => props => <div {...props} />);
jest.mock('../../PropertyItem', () => props => <div {...props} />);


const props = {
  displayProperties: {
    'Group 1': {
      properties: {
        foo: { display_value: 'Value of foo', title: 'Foo' },
        bar: { display_value: 'Value of bar', title: 'Bar' },
        foobar: { display_value: 'Value of foo and bar', title: 'Foo and Bar' },
      },
    },
  },
};


it('should render correctly', () => {
  let tree;
  renderer.act(() => {
    tree = renderer.create((
      <DefaultView
        {...props}
      />
    ));
  });
  expect(tree.toJSON()).toMatchSnapshot();
});
