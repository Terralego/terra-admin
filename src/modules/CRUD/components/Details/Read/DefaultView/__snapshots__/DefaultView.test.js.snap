// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<ul>
  <li>
    <span>
      CRUD.details.other
    </span>
    <div
      editProperties={Object {}}
      properties={
        Object {
          "bar": Object {
            "display_value": "Value of bar",
            "title": "Bar",
          },
          "foo": Object {
            "display_value": "Value of foo",
            "title": "Foo",
          },
          "foobar": Object {
            "display_value": "Value of foo and bar",
            "title": "Foo and Bar",
          },
        }
      }
    />
  </li>
</ul>
`;
